### PR TITLE
HEEDLS-409 expand boolean parse options for ObjectiveCcArray

### PR DESCRIPTION
**Pull request by JBsoftwire**
### JIRA link
[_HEEDLS-409_](https://softwiretech.atlassian.net/browse/HEEDLS-409)

### Description
The endpoint needs to interpret various strings (numerical 1 and 0 as well as 'true' and 'false') as booleans. I've added a special handler in the tracker service to do this, and relevant tests.

### Screenshots
N/A

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Could do with an extra unit test. Think there was a typo in the accepted int strings.

**Review by DanBloxham-sw — APPROVED** [review]
I think this is good now.

File: DigitalLearningSolutions.Data/Services/TrackerService.cs [hunk 1]
```diff
@@ -63,6 +64,24 @@ public string ProcessQuery(TrackerEndpointQueryParams query)
             }
         }
 
+        private bool? ConvertToNullableBoolean(string? value)
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Data/Services/TrackerService.cs:67]: The way this converts strings to booleans is quite specific to tracker endpoint query params. Could be worth making that slightly more explicit in the title. E.g. Convert*Param*ToNullableBoolean.

File: DigitalLearningSolutions.Data.Tests/Services/TrackerServiceTests.cs [hunk 2]
```diff
@@ -107,17 +107,29 @@ public void ProcessQuery_with_valid_action_correctly_serialises_null_response()
         }
 
         [Test]
-        public void ProcessQuery_with_GetObjectiveArrayCc_action_passes_query_params()
+        [TestCase("true", true)]
+        [TestCase("True", true)]
+        [TestCase("false", false)]
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Data.Tests/Services/TrackerServiceTests.cs:112]: Bah, missed this. We don't have a "False" case, so we may as well add that in. You can merge without rereview after that though.

**Review by SteveJacksonSoft — APPROVED** [review]

**Pull request merged**